File: frontend/src/utils/workforceStatus.js
/**
 * Canonical AI Workforce Status Contract (Frontend Mirror)
 * Defined in accordance with Phase 0 — Task 0.9.
 * Perfectly mirrors backend/internal/aitasks/workforce_status.go
 */

export const WORKFORCE_STATUS = {
  QUEUED: 'queued',
  CLAIMED: 'claimed',
  PROCESSING: 'processing',
  WAITING_FOR_APPROVAL: 'waiting_for_approval',
  PAUSED: 'paused',
  RETRYING: 'retrying',
  COMPLETED: 'completed',
  COMPLETED_WITH_FAILOVER: 'completed_with_failover',
  COMPLETED_IN_MOCK_MODE: 'completed_in_mock_mode',
  FAILED: 'failed',
  CANCELLED: 'cancelled',
  STALE: 'stale',
  UNKNOWN: 'unknown',
};

export const WORKFORCE_STATUS_METADATA = {
  [WORKFORCE_STATUS.QUEUED]: {
    value: WORKFORCE_STATUS.QUEUED,
    label: 'Queued',
    description: 'Task is waiting in the operational queue for a worker to claim it.',
    severity: 'neutral',
    color: '#475569',
    bg: '#f1f5f9',
    border: '#cbd5e1',
    isTerminal: false,
    isActionable: true,
    retryPermitted: false,
    approvalRequired: false,
    pulse: true,
  },
  [WORKFORCE_STATUS.CLAIMED]: {
    value: WORKFORCE_STATUS.CLAIMED,
    label: 'Claimed',
    description: 'Task claimed by a worker process and preparing execution.',
    severity: 'info',
    color: '#0284c7',
    bg: '#e0f2fe',
    border: '#bae6fd',
    isTerminal: false,
    isActionable: true,
    retryPermitted: false,
    approvalRequired: false,
    pulse: true,
  },
  [WORKFORCE_STATUS.PROCESSING]: {
    value: WORKFORCE_STATUS.PROCESSING,
    label: 'Processing',
    description: 'Agent workflow is actively executing nodes and tools.',
    severity: 'info',
    color: '#2563eb',
    bg: '#eff6ff',
    border: '#bfdbfe',
    isTerminal: false,
    isActionable: true,
    retryPermitted: false,
    approvalRequired: false,
    pulse: true,
  },
  [WORKFORCE_STATUS.WAITING_FOR_APPROVAL]: {
    value: WORKFORCE_STATUS.WAITING_FOR_APPROVAL,
    label: 'Awaiting Sign-Off',
    description: 'Workflow paused at a HITL interrupt awaiting human sign-off.',
    severity: 'warning',
    color: '#b45309',
    bg: '#fef3c7',
    border: '#fde68a',
    isTerminal: false,
    isActionable: true,
    retryPermitted: false,
    approvalRequired: true,
    pulse: false,
  },
  [WORKFORCE_STATUS.PAUSED]: {
    value: WORKFORCE_STATUS.PAUSED,
    label: 'Paused',
    description: 'Workflow execution is paused pending external dependencies.',
    severity: 'neutral',
    color: '#475569',
    bg: '#f1f5f9',
    border: '#cbd5e1',
    isTerminal: false,
    isActionable: true,
    retryPermitted: false,
    approvalRequired: false,
    pulse: false,
  },
  [WORKFORCE_STATUS.RETRYING]: {
    value: WORKFORCE_STATUS.RETRYING,
    label: 'Retrying',
    description: 'Encountered transient fault and is scheduled for backoff retry.',
    severity: 'warning',
    color: '#c2410c',
    bg: '#fff7ed',
    border: '#fed7aa',
    isTerminal: false,
    isActionable: true,
    retryPermitted: false,
    approvalRequired: false,
    pulse: true,
  },
  [WORKFORCE_STATUS.COMPLETED]: {
    value: WORKFORCE_STATUS.COMPLETED,
    label: 'Completed',
    description: 'Task executed to completion with business output committed.',
    severity: 'success',
    color: '#047857',
    bg: '#ecfdf5',
    border: '#a7f3d0',
    isTerminal: true,
    isActionable: false,
    retryPermitted: false,
    approvalRequired: false,
    pulse: false,
  },
  [WORKFORCE_STATUS.COMPLETED_WITH_FAILOVER]: {
    value: WORKFORCE_STATUS.COMPLETED_WITH_FAILOVER,
    label: 'Completed (Failover)',
    description: 'Task completed successfully using the secondary failover provider.',
    severity: 'success',
    color: '#0f766e',
    bg: '#f0fdfa',
    border: '#99f6e4',
    isTerminal: true,
    isActionable: false,
    retryPermitted: false,
    approvalRequired: false,
    pulse: false,
  },
  [WORKFORCE_STATUS.COMPLETED_IN_MOCK_MODE]: {
    value: WORKFORCE_STATUS.COMPLETED_IN_MOCK_MODE,
    label: 'Completed (Mock)',
    description: 'Task simulated via deterministic mock engine in development mode.',
    severity: 'neutral',
    color: '#6d28d9',
    bg: '#f5f3ff',
    border: '#ddd6fe',
    isTerminal: true,
    isActionable: false,
    retryPermitted: false,
    approvalRequired: false,
    pulse: false,
  },
  [WORKFORCE_STATUS.FAILED]: {
    value: WORKFORCE_STATUS.FAILED,
    label: 'Failed',
    description: 'Task failed and exhausted retries, or suffered fatal error.',
    severity: 'danger',
    color: '#b91c1c',
    bg: '#fef2f2',
    border: '#fecaca',
    isTerminal: true,
    isActionable: true,
    retryPermitted: true,
    approvalRequired: false,
    pulse: false,
  },
  [WORKFORCE_STATUS.CANCELLED]: {
    value: WORKFORCE_STATUS.CANCELLED,
    label: 'Cancelled',
    description: 'Task was cancelled by an operator before final execution.',
    severity: 'neutral',
    color: '#64748b',
    bg: '#f8fafc',
    border: '#e2e8f0',
    isTerminal: true,
    isActionable: false,
    retryPermitted: false,
    approvalRequired: false,
    pulse: false,
  },
  [WORKFORCE_STATUS.STALE]: {
    value: WORKFORCE_STATUS.STALE,
    label: 'Stale / Interrupted',
    description: 'Worker heartbeat expired or task lease exceeded without progress.',
    severity: 'danger',
    color: '#be123c',
    bg: '#fff1f2',
    border: '#fecdd3',
    isTerminal: false,
    isActionable: true,
    retryPermitted: true,
    approvalRequired: false,
    pulse: true,
  },
  [WORKFORCE_STATUS.UNKNOWN]: {
    value: WORKFORCE_STATUS.UNKNOWN,
    label: 'Unknown',
    description: 'Source execution data is genuinely unavailable or unmapped.',
    severity: 'neutral',
    color: '#64748b',
    bg: '#f8fafc',
    border: '#e2e8f0',
    isTerminal: false,
    isActionable: false,
    retryPermitted: false,
    approvalRequired: false,
    pulse: false,
  },
};

/**
 * Normalizes any backend, database, or graph status string into the canonical contract.
 */
export function normalizeWorkforceStatus(rawStatus, task = {}) {
  if (!rawStatus) return WORKFORCE_STATUS.UNKNOWN;

  const s = String(rawStatus).toLowerCase().trim();

  // Handle direct matches
  if (WORKFORCE_STATUS_METADATA[s]) {
    return s;
  }

  // Handle mock mode
  if (task.mock_mode || task.is_mock) {
    if (s === 'completed' || s === 'succeeded' || s === 'success') {
      return WORKFORCE_STATUS.COMPLETED_IN_MOCK_MODE;
    }
  }

  // Handle failover
  if (task.provider_failover || task.used_failover) {
    if (s === 'completed' || s === 'succeeded' || s === 'success') {
      return WORKFORCE_STATUS.COMPLETED_WITH_FAILOVER;
    }
  }

  // Map legacy / graph states
  switch (s) {
    case 'queued':
    case 'pending':
    case 'submitted':
      return WORKFORCE_STATUS.QUEUED;
    case 'claimed':
      return WORKFORCE_STATUS.CLAIMED;
    case 'processing':
    case 'running':
    case 'in_progress':
    case 'collecting_information':
    case 'analyzing_data':
    case 'waiting_for_llm':
    case 'generating_draft':
      return WORKFORCE_STATUS.PROCESSING;
    case 'waiting_for_approval':
    case 'waiting_for_human_review':
    case 'approval_required':
      return WORKFORCE_STATUS.WAITING_FOR_APPROVAL;
    case 'paused':
      return WORKFORCE_STATUS.PAUSED;
    case 'retrying':
      return WORKFORCE_STATUS.RETRYING;
    case 'completed':
    case 'succeeded':
    case 'success':
      return WORKFORCE_STATUS.COMPLETED;
    case 'failed':
    case 'error':
    case 'rejected':
      return WORKFORCE_STATUS.FAILED;
    case 'cancelled':
    case 'aborted':
      return WORKFORCE_STATUS.CANCELLED;
    case 'stale':
    case 'interrupted':
    case 'timed_out':
      return WORKFORCE_STATUS.STALE;
    default:
      return WORKFORCE_STATUS.UNKNOWN;
  }
}

/**
 * Helper to fetch metadata with fallback to UNKNOWN.
 */
export function getWorkforceStatusMetadata(rawStatus, task = {}) {
  const norm = normalizeWorkforceStatus(rawStatus, task);
  return WORKFORCE_STATUS_METADATA[norm] || WORKFORCE_STATUS_METADATA[WORKFORCE_STATUS.UNKNOWN];
}

File: backend/internal/aitasks/workforce_status.go
package aitasks

import (
	"fmt"
	"strings"
	"time"
)

// Authoritative 13 Workforce Status Constants
const (
	WorkforceStatusQueued                = "queued"
	WorkforceStatusClaimed               = "claimed"
	WorkforceStatusProcessing            = "processing"
	WorkforceStatusWaitingForApproval    = "waiting_for_approval"
	WorkforceStatusPaused                = "paused"
	WorkforceStatusRetrying              = "retrying"
	WorkforceStatusCompleted             = "completed"
	WorkforceStatusCompletedWithFailover = "completed_with_failover"
	WorkforceStatusCompletedInMockMode   = "completed_in_mock_mode"
	WorkforceStatusFailed                = "failed"
	WorkforceStatusCancelled             = "cancelled"
	WorkforceStatusStale                 = "stale"
	WorkforceStatusUnknown               = "unknown"
)

// WorkforceStatusMetadata provides safe, standardized display and action semantics for each state.
type WorkforceStatusMetadata struct {
	Value            string `json:"value"`
	Label            string `json:"label"`
	Description      string `json:"description"`
	Severity         string `json:"severity"` // "info", "warning", "success", "error", "neutral", "purple"
	IsTerminal       bool   `json:"is_terminal"`
	IsActionable     bool   `json:"is_actionable"`
	RetryPermitted   bool   `json:"retry_permitted"`
	ApprovalRequired bool   `json:"approval_required"`
}

// Canonical status registry definitions
var statusMetadataRegistry = map[string]WorkforceStatusMetadata{
	WorkforceStatusQueued: {
		Value:            WorkforceStatusQueued,
		Label:            "Queued",
		Description:      "Task is staged in queue waiting for an available worker.",
		Severity:         "neutral",
		IsTerminal:       false,
		IsActionable:     true, // Can cancel
		RetryPermitted:   false,
		ApprovalRequired: false,
	},
	WorkforceStatusClaimed: {
		Value:            WorkforceStatusClaimed,
		Label:            "Claimed",
		Description:      "Task has been leased by an active worker thread and is initializing.",
		Severity:         "info",
		IsTerminal:       false,
		IsActionable:     false,
		RetryPermitted:   false,
		ApprovalRequired: false,
	},
	WorkforceStatusProcessing: {
		Value:            WorkforceStatusProcessing,
		Label:            "Processing",
		Description:      "Agent is actively executing workflows, calling tools, or querying rates.",
		Severity:         "info",
		IsTerminal:       false,
		IsActionable:     true, // Can cancel
		RetryPermitted:   false,
		ApprovalRequired: false,
	},
	WorkforceStatusWaitingForApproval: {
		Value:            WorkforceStatusWaitingForApproval,
		Label:            "Waiting for Sign-Off",
		Description:      "Autonomous execution is paused at a governance gate awaiting human review.",
		Severity:         "purple",
		IsTerminal:       false,
		IsActionable:     true, // Can open approval modal
		RetryPermitted:   false,
		ApprovalRequired: true,
	},
	WorkforceStatusPaused: {
		Value:            WorkforceStatusPaused,
		Label:            "Paused",
		Description:      "Graph execution paused at an interrupt node.",
		Severity:         "warning",
		IsTerminal:       false,
		IsActionable:     true,
		RetryPermitted:   false,
		ApprovalRequired: true,
	},
	WorkforceStatusRetrying: {
		Value:            WorkforceStatusRetrying,
		Label:            "Retrying",
		Description:      "A transient provider error occurred; task is backed off for automatic retry.",
		Severity:         "warning",
		IsTerminal:       false,
		IsActionable:     true, // Can cancel
		RetryPermitted:   false,
		ApprovalRequired: false,
	},
	WorkforceStatusCompleted: {
		Value:            WorkforceStatusCompleted,
		Label:            "Completed",
		Description:      "AI work finished successfully with authorized actions persisted.",
		Severity:         "success",
		IsTerminal:       true,
		IsActionable:     false,
		RetryPermitted:   false,
		ApprovalRequired: false,
	},
	WorkforceStatusCompletedWithFailover: {
		Value:            WorkforceStatusCompletedWithFailover,
		Label:            "Completed (Failover)",
		Description:      "Task finished successfully after primary provider failed and switched to failover.",
		Severity:         "info",
		IsTerminal:       true,
		IsActionable:     false,
		RetryPermitted:   false,
		ApprovalRequired: false,
	},
	WorkforceStatusCompletedInMockMode: {
		Value:            WorkforceStatusCompletedInMockMode,
		Label:            "Completed (Mock Dev)",
		Description:      "Task finished in development test mode using deterministic mock response.",
		Severity:         "warning",
		IsTerminal:       true,
		IsActionable:     false,
		RetryPermitted:   false,
		ApprovalRequired: false,
	},
	WorkforceStatusFailed: {
		Value:            WorkforceStatusFailed,
		Label:            "Failed",
		Description:      "Execution failed due to unrecoverable error or exceeded maximum retry attempts.",
		Severity:         "error",
		IsTerminal:       true,
		IsActionable:     true, // Can manual retry
		RetryPermitted:   true,
		ApprovalRequired: false,
	},
	WorkforceStatusCancelled: {
		Value:            WorkforceStatusCancelled,
		Label:            "Cancelled",
		Description:      "Task was safely cancelled by operator or reviewer.",
		Severity:         "neutral",
		IsTerminal:       true,
		IsActionable:     false,
		RetryPermitted:   false,
		ApprovalRequired: false,
	},
	WorkforceStatusStale: {
		Value:            WorkforceStatusStale,
		Label:            "Stale / Needs Attention",
		Description:      "Worker lease expired without heartbeats; queued for automatic recovery or retry.",
		Severity:         "error",
		IsTerminal:       false,
		IsActionable:     true, // Can cancel or retry
		RetryPermitted:   true,
		ApprovalRequired: false,
	},
	WorkforceStatusUnknown: {
		Value:            WorkforceStatusUnknown,
		Label:            "Unknown",
		Description:      "Task status could not be authoritatively resolved from database.",
		Severity:         "neutral",
		IsTerminal:       false,
		IsActionable:     false,
		RetryPermitted:   false,
		ApprovalRequired: false,
	},
}

// GetStatusMetadata returns the canonical metadata descriptor for a given workforce status.
func GetStatusMetadata(status string) WorkforceStatusMetadata {
	norm := strings.ToLower(strings.TrimSpace(status))
	if meta, exists := statusMetadataRegistry[norm]; exists {
		return meta
	}
	return statusMetadataRegistry[WorkforceStatusUnknown]
}

// AgentWorkforceStat represents high-level metrics for one agent/workflow type.
type AgentWorkforceStat struct {
	AgentKey         string `json:"agent_key"`        // "pricing", "sales", "operations", etc.
	DisplayName      string `json:"display_name"`     // "Pricing Analyst"
	Module           string `json:"module"`           // "PRICING", "SALES", etc.
	ActiveTasks      int64  `json:"active_tasks"`
	Completed24h     int64  `json:"completed_24h"`
	Failed24h        int64  `json:"failed_24h"`
	WaitingApprovals int64  `json:"waiting_approvals"`
	Status           string `json:"status"`           // "ACTIVE", "IDLE", "ATTENTION", "OFFLINE"
}

// FailureCategoryStat aggregates recent failures by safe category.
type FailureCategoryStat struct {
	Category string `json:"category"`
	Count    int64  `json:"count"`
}

// WorkforceHealth exposes system readiness and component availability.
type WorkforceHealth struct {
	OverallStatus         string     `json:"overall_status"`          // "healthy", "degraded", "unavailable"
	BackendStatus         string     `json:"backend_status"`          // "healthy"
	SidecarStatus         string     `json:"sidecar_status"`          // "healthy", "unavailable"
	WorkerStatus          string     `json:"worker_status"`           // "healthy", "degraded", "idle"
	QueueStatus           string     `json:"queue_status"`            // "healthy", "lagging", "blocked"
	CheckpointStatus      string     `json:"checkpoint_status"`       // "healthy", "degraded"
	PrimaryProviderStatus string     `json:"primary_provider_status"` // "healthy", "degraded"
	FailoverStatus        string     `json:"failover_status"`         // "ready", "degraded"
	LastWorkerHeartbeat   *time.Time `json:"last_worker_heartbeat,omitempty"`
}

// WorkforceSummary aggregates live metrics for an authenticated organization.
type WorkforceSummary struct {
	OrgID                   int64                         `json:"org_id"`
	TotalActiveTasks        int64                         `json:"total_active_tasks"`
	QueuedTasks             int64                         `json:"queued_tasks"`
	ProcessingTasks         int64                         `json:"processing_tasks"`
	WaitingForApprovalTasks int64                         `json:"waiting_for_approval_tasks"`
	RetryingTasks           int64                         `json:"retrying_tasks"`
	FailedTasks             int64                         `json:"failed_tasks"`
	StaleTasks              int64                         `json:"stale_tasks"`
	CompletedRecent24h      int64                         `json:"completed_recent_24h"`
	CompletedWithFailover   int64                         `json:"completed_with_failover_24h"`
	CompletedInMockMode     int64                         `json:"completed_in_mock_mode_24h"`
	AvgDurationMs           int64                         `json:"avg_duration_ms"`
	ActiveAgentsCount       int                           `json:"active_agents_count"`
	ByAgent                 map[string]AgentWorkforceStat `json:"by_agent"`
	RecentFailures          []FailureCategoryStat         `json:"recent_failures"`
	Health                  WorkforceHealth               `json:"health"`
	LastUpdated             time.Time                     `json:"last_updated"`
}

// WorkforceTaskItem is a safe, enriched representation of an AI task for operational views.
type WorkforceTaskItem struct {
	ID            int64                   `json:"id"`
	OrgID         int64                   `json:"org_id"`
	TaskType      string                  `json:"task_type"`
	AgentKey      string                  `json:"agent_key"`
	AgentName     string                  `json:"agent_name"`
	Module        string                  `json:"module"`
	RelatedRef    string                  `json:"related_ref"`
	RelatedID     string                  `json:"related_id"`
	Status        string                  `json:"status"` // Authoritative 13-status value
	StatusMeta    WorkforceStatusMetadata `json:"status_meta"`
	CreatedAt     time.Time               `json:"created_at"`
	UpdatedAt     time.Time               `json:"updated_at"`
	StartedAt     *time.Time              `json:"started_at,omitempty"`
	CompletedAt   *time.Time              `json:"completed_at,omitempty"`
	DurationMs    *int64                  `json:"duration_ms,omitempty"`
	RetryCount    int                     `json:"retry_count"`
	MaxRetries    int                     `json:"max_retries"`
	IsStale       bool                    `json:"is_stale"`
	ApprovalID    *int64                  `json:"approval_id,omitempty"`
	ErrorCategory *string                 `json:"error_category,omitempty"`
	ErrorMessage  *string                 `json:"error_message,omitempty"` // Secret-redacted
	CanRetry      bool                    `json:"can_retry"`
	CanCancel     bool                    `json:"can_cancel"`
	NavigationURL string                  `json:"navigation_url"`
	ThreadID      *string                 `json:"thread_id,omitempty"`
	CorrelationID *string                 `json:"correlation_id,omitempty"`
}

// WorkforceTaskFilter specifies filtering criteria for querying AI workforce tasks.
type WorkforceTaskFilter struct {
	Status           string `json:"status,omitempty"`
	AgentKey         string `json:"agent_key,omitempty"`
	Module           string `json:"module,omitempty"`
	RequiresApproval *bool  `json:"requires_approval,omitempty"`
	FailedOrStale    *bool  `json:"failed_or_stale,omitempty"`
	Search           string `json:"search,omitempty"`
	Limit            int    `json:"limit,omitempty"`
	Offset           int    `json:"offset,omitempty"`
}

// MapTaskTypeToAgent maps raw task_type strings to canonical agent identity, module, and navigation.
func MapTaskTypeToAgent(taskType string) (agentKey, agentName, module, defaultNav string) {
	upper := strings.ToUpper(strings.TrimSpace(taskType))
	switch {
	case strings.Contains(upper, "PRICING"):
		return "pricing", "Pricing Analyst", "PRICING", "/dashboard/rfqs"
	case strings.Contains(upper, "EMAIL") || strings.Contains(upper, "SALES"):
		return "sales", "Sales Email Parser", "SALES", "/dashboard/leads"
	case strings.Contains(upper, "CARRIER") || strings.Contains(upper, "OPERATIONS") || strings.Contains(upper, "TRACKING"):
		return "operations", "Operations Carrier Tracker", "OPERATIONS", "/dashboard/shipments"
	case strings.Contains(upper, "CONTRACT") || strings.Contains(upper, "DOC_PROCESS") || strings.Contains(upper, "RATE_EXTRACT"):
		return "contracts", "Contracts Intelligence", "CONTRACTS", "/dashboard/contracts"
	case strings.Contains(upper, "COMPLIANCE") || strings.Contains(upper, "DOC_VERIFY"):
		return "compliance", "Compliance Auditor", "COMPLIANCE", "/dashboard/shipments"
	case strings.Contains(upper, "FINANCE") || strings.Contains(upper, "BILL_RECONCILE") || strings.Contains(upper, "INVOICE"):
		return "finance", "Finance Invoice Auditor", "FINANCE", "/dashboard/invoices"
	case strings.Contains(upper, "LEAD"):
		return "leads", "Lead Scoring Specialist", "LEADS", "/dashboard/leads"
	case strings.Contains(upper, "OUTREACH"):
		return "outreach", "Outreach Copywriter", "OUTREACH", "/dashboard/outreach"
	default:
		return "general", "General AI Assistant", "OPERATIONS", "/dashboard"
	}
}

// ResolveTaskWorkforceStatus maps a task row and execution telemetry into one authoritative status.
func ResolveTaskWorkforceStatus(task *Task, isFailover bool, isMock bool) (string, WorkforceStatusMetadata) {
	if task == nil {
		return WorkforceStatusUnknown, GetStatusMetadata(WorkforceStatusUnknown)
	}

	upperStatus := strings.ToUpper(strings.TrimSpace(task.Status))

	// 1. Check for Stale processing condition (lease expired without heartbeat)
	if upperStatus == StatusProcessing && task.LeaseExpiresAt != nil && task.LeaseExpiresAt.Before(time.Now()) {
		return WorkforceStatusStale, GetStatusMetadata(WorkforceStatusStale)
	}

	// 2. Map standard task states
	switch upperStatus {
	case StatusQueued:
		if task.WorkerID != nil && *task.WorkerID != "" {
			return WorkforceStatusClaimed, GetStatusMetadata(WorkforceStatusClaimed)
		}
		return WorkforceStatusQueued, GetStatusMetadata(WorkforceStatusQueued)

	case StatusProcessing:
		return WorkforceStatusProcessing, GetStatusMetadata(WorkforceStatusProcessing)

	case StatusWaitingForApproval:
		return WorkforceStatusWaitingForApproval, GetStatusMetadata(WorkforceStatusWaitingForApproval)

	case StatusRetrying:
		return WorkforceStatusRetrying, GetStatusMetadata(WorkforceStatusRetrying)

	case StatusCompleted:
		if isMock {
			return WorkforceStatusCompletedInMockMode, GetStatusMetadata(WorkforceStatusCompletedInMockMode)
		}
		if isFailover {
			return WorkforceStatusCompletedWithFailover, GetStatusMetadata(WorkforceStatusCompletedWithFailover)
		}
		return WorkforceStatusCompleted, GetStatusMetadata(WorkforceStatusCompleted)

	case StatusFailed:
		return WorkforceStatusFailed, GetStatusMetadata(WorkforceStatusFailed)

	case StatusCancelled, StatusRejected:
		return WorkforceStatusCancelled, GetStatusMetadata(WorkforceStatusCancelled)

	default:
		return WorkforceStatusUnknown, GetStatusMetadata(WorkforceStatusUnknown)
	}
}

// ResolveRelatedRef resolves display label, related entity ID, and safe navigation link.
func ResolveRelatedRef(task *Task) (label, id, navURL string) {
	if task == nil {
		return "Unknown Task", "", "/dashboard"
	}

	// Check EntityType and EntityID first
	if task.EntityType != nil && task.EntityID != nil && *task.EntityID != "" {
		eType := strings.ToLower(strings.TrimSpace(*task.EntityType))
		eID := *task.EntityID
		switch eType {
		case "rfq":
			return fmt.Sprintf("RFQ #%s", eID), eID, fmt.Sprintf("/dashboard/rfqs/%s", eID)
		case "lead":
			return fmt.Sprintf("Lead #%s", eID), eID, fmt.Sprintf("/dashboard/leads/%s", eID)
		case "shipment":
			return fmt.Sprintf("Shipment #%s", eID), eID, fmt.Sprintf("/dashboard/shipments/%s", eID)
		case "invoice":
			return fmt.Sprintf("Invoice #%s", eID), eID, "/dashboard/invoices"
		case "contract", "document":
			return fmt.Sprintf("Contract #%s", eID), eID, fmt.Sprintf("/dashboard/contracts/%s", eID)
		default:
			return fmt.Sprintf("%s #%s", strings.ToUpper(eType), eID), eID, "/dashboard"
		}
	}

	// Check ResourceID
	if task.ResourceID != nil && *task.ResourceID > 0 {
		_, _, _, defNav := MapTaskTypeToAgent(task.TaskType)
		return fmt.Sprintf("Record #%d", *task.ResourceID), fmt.Sprintf("%d", *task.ResourceID), defNav
	}

	// Check DocumentID
	if task.DocumentID != nil && *task.DocumentID != "" {
		return fmt.Sprintf("Doc #%s", *task.DocumentID), *task.DocumentID, "/dashboard/contracts"
	}

	// Fallback to Task ID
	_, _, _, defNav := MapTaskTypeToAgent(task.TaskType)
	return fmt.Sprintf("Task #%d", task.ID), fmt.Sprintf("%d", task.ID), defNav
}
